scripts/DRL_wandb.py

```diff
@@ -0,0 +1,410 @@
+                vf_coef=args.vf_coef,
+                max_grad_norm=args.max_grad_norm,
+                seed=args.seed,
+                )
+#--------------------------------------------------------#
+#                           SAC                          #
+#--------------------------------------------------------#
+elif args.algorithm == 'SAC':
+    model = SAC(policy='MlpPolicy',
+                env=env,
+                seed=args.seed,
+                learning_rate=args.learning_rate,
+                buffer_size=args.buffer_size,
+                batch_size=args.batch_size,
+                tau=args.tau,
+                gamma=args.gamma,
+                )
+#--------------------------------------------------------#
+#                           TD3                          #
+#--------------------------------------------------------#
+elif args.algorithm == 'TD3':
+    model = TD3(policy='MlpPolicy',
+                env=env, seed=args.seed,
+                learning_rate=args.learning_rate,
+                buffer_size=args.buffer_size,
+                batch_size=args.batch_size,
+                tau=args.tau,
+                gamma=args.gamma,
+                train_freq=(1, 'episode'),
+                action_noise=None,
+                replay_buffer_class=None,
+                replay_buffer_kwargs=None,
+                optimize_memory_usage=False,
+                policy_delay=2,
+                target_policy_noise=0.2,
+                target_noise_clip=0.5,
+                create_eval_env=False,
+                policy_kwargs=None,
+                verbose=0,
+                device='auto',
+                _init_setup_model=True)
+#--------------------------------------------------------#
+#                           Error                        #
+#--------------------------------------------------------#
+else:
+    raise RuntimeError(
+        F'Algorithm specified [{args.algorithm}] is not registered.')
+
+#--------------------------------------------------------#
+#                     Set Loggers                        #
+#--------------------------------------------------------#
+logger = Logger(folder="runs/", output_formats=[WandBOutputFormat()])
+model.set_logger(logger)
+
+
+# ---------------------------------------------------------------------------- #
+#       Calculating total training timesteps based on number of episodes       #
+# ---------------------------------------------------------------------------- #
+n_timesteps_episode = env.simulator._eplus_one_epi_len / \
+    env.simulator._eplus_run_stepsize
+timesteps = args.episodes * n_timesteps_episode - 1
+
+# ---------------------------------------------------------------------------- #
+#                                   CALLBACKS                                  #
+# ---------------------------------------------------------------------------- #
+callbacks = []
+
+# Set up Evaluation and saving best model
+if args.evaluation:
```

Comment:
You can decide to have the log activated at the same time as you do evaluations every few episodes, so this if-else is not appropriate, is it?

Reply:
That is correct. I haven't tried to check if `LoggerEvalCallback` works with WandB. I am trying to figure out why you are unable to log for training while I am. Once we fix that issue, I will make all the changes you have suggested!